### PR TITLE
add reCaptcha

### PR DESCRIPTION
**Pull request by vmcodes**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [**94**](https://github.com/Web-Dev-Path/web-dev-path/issues/94) |

#### Have you updated the CHANGELOG.md file? If not, please do it.
Yes

#### What is this change?
Added reCaAPTCHA to the NewsletterForm.js component, as well as the register.js api function.

#### Were there any complications while making this change?
No

#### How did you verify this change?
Tested locally and was able to see the the proper response from reCAPTCHA logged when the form fields were valid. I was also able to receive the reCAPTCHA test on occasion, which confirmed that reCAPTCHA was working.

#### When should this be merged?
This should be able to be merged after review.

### CODE REVIEW

File: components/mailchimp/NewsletterForm.js [hunk 1]
```diff
@@ -77,22 +124,36 @@ const NewsletterForm = ({ status, message, onValidated }) => {
           >
             <input
               className={`${newsletterStyles.newsletter__input} ${newsletterStyles.newsletter__name}`}
+              onChange={event => setName(event?.target?.value ?? '')}
```
Comment (mariana-caldas) [inline components/mailchimp/NewsletterForm.js:127]: Thanks for adding a check for the name as well, @vmcodes !

**Review by mariana-caldas — APPROVED** [review]
LGTM, @vmcodes ! 🚀 🎉 
Could you pls just wait a couple of days before merging it so we give others a chance to review it too? Thanks!

**Comment by cherylli** [comment]
Not sure if it's my local issue here but I tried maybe 10 times and only 2 times the reCaptcha shows up. The rest of the time it only shows `Almost finished... We need to confirm your email address. To complete the subscription process, please click the link in the email we just sent you.`

I have tried entering a new email address and restarted the server.

Also got this once but cannot recreate, but it looks like google error.
[](https://i.imgur.com/bMIN9Cr.png)

**Comment by mariana-caldas** [comment]
> Not sure if it's my local issue here but I tried maybe 10 times and only 2 times the reCaptcha shows up. The rest of the time it only shows `Almost finished... We need to confirm your email address. To complete the subscription process, please click the link in the email we just sent you.`
> 
> I have tried entering a new email address and restarted the server.
> 
> Also got this once but cannot recreate, but it looks like google error. [](https://i.imgur.com/bMIN9Cr.png)

Thanks for testing it locally thoroughly, @cherylli ! ⭐ 
The reCaptcha API "talks" to the browsers and the challenge is only triggered when the browsers "find" the requests suspicious so that number seems pretty normal to me. 
That error seems to be random as well, so I wouldn't worry about it for now.
Now, you can really review the codebase by checking the code consistency, variable naming, or if the code is repetitive (not DRY) and needs refactoring with this tutorial:
https://www.youtube.com/watch?v=lSnbOtw4izI

**Review by cherylli — APPROVED** [review]
looks good

**Pull request merged**